File: src/com/suushiemaniac/lang/json/value/JsonBoolean.java
package com.suushiemaniac.lang.json.value;

import com.suushiemaniac.lang.json.JSON;

public class JsonBoolean extends JsonType {
    private boolean value;

    public JsonBoolean(boolean value) {
        this(null, value);
    }

    public JsonBoolean(JSON parent, boolean value) {
        super(parent);
        this.value = value;
    }

    @Override
    public boolean booleanValue() {
        return this.value;
    }

    @Override
    public String toString() {
        return String.valueOf(this.value);
    }

    @Override
    public boolean equals(Object other) {
        return other instanceof JsonBoolean && ((JsonBoolean) other).value == this.value;
    }

    @Override
    public void set(JSON value) {
        if (value instanceof JsonBoolean) this.value = value.booleanValue();
    }
}

File: src/main/kotlin/com/suushiemaniac/lang/json/value/JsonBoolean.kt
package com.suushiemaniac.lang.json.value

import com.suushiemaniac.lang.json.JSON

class JsonBoolean(parent: JSON?, private var value: Boolean) : JsonType(parent) {
    constructor(value: Boolean) : this(null, value)

    override fun booleanValue(): Boolean {
        return this.value
    }

    override fun stringValue(): String {
        return "" + this.value
    }

    override fun toNative(): Any {
        return this.booleanValue()
    }

    override fun intValue(): Int {
        return if (this.value) 1 else 0
    }

    override fun floatValue(): Float {
        return if (this.value) 1f else 0f
    }

    override fun toString(): String {
        return this.value.toString()
    }

    override fun equals(other: Any?): Boolean {
        return other is JsonBoolean && other.value == this.value
    }

    override fun hashCode(): Int {
        return this.value.hashCode()
    }

    override fun set(value: JSON) {
        if (value is JsonBoolean) this.value = value.booleanValue()
    }
}